api/src/routing.js

```diff
@@ -111,6 +112,16 @@ app.postJsonOrCsv = (path, callback) => handleJsonOrCsvRequest('post', path, cal
 app.postJson = (path, callback) => handleJsonRequest('post', path, callback);
 app.putJson = (path, callback) => handleJsonRequest('put', path, callback);
 
+app.use(prometheusMiddleware({
+  metricsPath: '/api_prometheus_metrics',
+  // based on one-month analysed period of production traffic
+  // each durationBuckets captures ~5% of traffic unless otherwise indicated
+  durationBuckets: [
```

Comment:
Seeking a thought partner here

Prometheus stores values in histograms. All values are approximated into these buckets and so we need to understand the distribution of our data to get meaningful useful output.

I analysed the total_time spent in requests for 23 days of production traffic for Muso-Mali (1.2M requests). This is generally regarded as a "healthy instance" which is well used and loved. I grouped and counted them all by the closest second. [Result is here](https://docs.google.com/spreadsheets/d/1tpennYg1H5vGXK_hFTf8mfAjB2U-m3Lr8jf69PrTaJU/edit#gid=2042819910).

I suggest here to make 20 buckets with data evenly distributed across the buckets. To cater to newer projects, I've split the first bucket to be more granular.

Yes: this is saying that 5% of production traffic currently spends >8 hours at our API tier.

Reply:
I think this is a good starting point, especially to get some data and reassess later. 
Having >8h requests seems very strange, since our ALB timeout is set to 1h. These requests won't serve anyone. 

The other option I'm thinking is to have less buckets and think Gaussian curve, with ~6 buckets: 
![image](https://github.com/medic/cht-core/assets/35681649/098921ae-b7ee-4283-b237-9e83c403a169)

Reply:
I like the idea to fit our data to a distribution curve. [Sum of erlang distributions](https://newrelic.com/blog/best-practices/expected-distributions-website-response-times) would be a good option. Another [good read here](https://newrelic.com/blog/best-practices/performance-metrics-in-time-series-data) if we ever want to model this instead of basing off production data.

I suspect six buckets is too few. The default for the library is to have 9 buckets just monitoring values from 0 to 0.5 seconds.   We are interested in a range of response times which are 10000x larger so I suspect ~25 buckets is not inappropriate.  Part of me is actually inclined toward more buckets since the cost is not high and our understanding of this is still quite low.

Reply:
Opened https://github.com/medic/cht-core/issues/8372 for requests > 8hr

Reply:
I think we should go with your suggestion of 38 buckets. 
> our understanding of response durations is still quite low.

I agree.

Reply:
Apdex monitoring [here](https://github.com/medic/cht-watchdog/pull/75#issue-1809492389) is one more factor to consider when setting these buckets. Currently using 90 and 180 for it.